gramex/data.py

```diff
@@ -1460,9 +1464,24 @@ def _mongodb_json(obj):
 def _filter_mongodb(url, controls, args, database=None, collection=None, query=None, **kwargs):
     '''TODO: Document function and usage'''
     table = _mongodb_collection(url, database, collection, **kwargs)
+
+    for key, value in args.items():
+        if key == '_id':
+            args[key] = list(map(ObjectId, value))
+            continue
+
+        for val in value:
+            if not val.isnumeric():
+                break
+        else:
+            args[key] = list(map(int, value))
+
```

Comment:
You can get the first 100 rows from `meta_cols = pd.DataFrame(table.find().limit(100))`. Use the DataFrame dtype to figure out the type and convert based on that.

There's a line that goes `convert = int if (meta_cols[col].dtype == pd.np.int64) else meta_cols[col].dtype.type`

If you plug into that line and extend it for these scenarios, we should be OK